spec/requests/courses_spec.rb

```diff
@@ -1,19 +1,73 @@
 require 'helpers/rails_helper'
+require 'helpers/api_helper'
 
 RSpec.describe "Courses", type: :request do
   include_context "with authenticated requests"
-  let(:user) { create(:user) }
+  let(:student) { create(:user) }
+  let(:teacher) { create(:user, :teacher) }
   let!(:course) { create(:course) }
 
-  before do
```

Comment:
Alternatively, in the before block, we could authenticate a student and a teacher and just work with those in the `describe`s.